charger/ocpp.go

```diff
@@ -165,107 +170,121 @@ func NewOCPP(id string, connector int, idtag string,
-					return cmp.Compare(i.Key, j.Key)
-				})
+				c.log.DEBUG.Printf("%s (%s): %s", opt.Key, rw[opt.Readonly], *opt.Value)
+
+				switch opt.Key {
+				case ocpp.KeyChargeProfileMaxStackLevel:
+					if val, err := strconv.Atoi(*opt.Value); err == nil {
+						c.stackLevel = val
+					}
+
+				case ocpp.KeyChargingScheduleAllowedChargingRateUnit:
+					if *opt.Value == "Power" || *opt.Value == "W" { // "W" is not allowed by spec but used by some CPs
+						c.chargingRateUnit = types.ChargingRateUnitWatts
+					}
 
-				rw := map[bool]string{false: "r/w", true: "r/o"}
+				case ocpp.KeyConnectorSwitch3to1PhaseSupported:
+					var val bool
+					if val, err = strconv.ParseBool(*opt.Value); err == nil {
+						c.phaseSwitching = val
+					}
 
-				for _, opt := range resp.ConfigurationKey {
-					if opt.Value == nil {
-						continue
+				case ocpp.KeyMaxChargingProfilesInstalled:
+					if val, err := strconv.Atoi(*opt.Value); err == nil {
+						c.chargingProfileId = val
 					}
 
-					c.log.TRACE.Printf("%s (%s): %s", opt.Key, rw[opt.Readonly], *opt.Value)
-
-					switch opt.Key {
-					case ocpp.KeyNumberOfConnectors:
-						var val int
-						if val, err = strconv.Atoi(*opt.Value); err == nil && connector > val {
-							err = fmt.Errorf("connector %d exceeds max available connectors: %d", connector, val)
-						}
-
-					case ocpp.KeyMeterValuesSampledData:
-						c.meterValuesSample = *opt.Value
-
-					case ocpp.KeyMeterValueSampleInterval:
-						var val int
-						if val, err = strconv.Atoi(*opt.Value); err == nil {
-							meterSampleInterval = time.Duration(val) * time.Second
-						}
-
-					case ocpp.KeyConnectorSwitch3to1PhaseSupported:
-						var val bool
-						if val, err = strconv.ParseBool(*opt.Value); err == nil {
-							c.phaseSwitching = val
-						}
-
-					case ocpp.KeyAlfenPlugAndChargeIdentifier:
-						if c.idtag == defaultIdTag {
-							c.idtag = *opt.Value
-							c.log.DEBUG.Printf("overriding default `idTag` with Alfen-specific value: %s", c.idtag)
-						}
-
-					case ocpp.KeyChargingScheduleAllowedChargingRateUnit:
-						if *opt.Value == "W" || *opt.Value == "Power" {
-							c.chargingRateUnit = types.ChargingRateUnitWatts
-						}
+				case ocpp.KeyMeterValuesSampledData:
+					MeterValuesSampledData = !opt.Readonly
+				case ocpp.KeyMeterValuesSampledDataMaxLength:
+					if val, err := strconv.Atoi(*opt.Value); err == nil {
+						MeterValuesSampledDataMaxLength = val
 					}
 
-					if err != nil {
-						break
+				case ocpp.KeyNumberOfConnectors:
+					var val int
+					if val, err = strconv.Atoi(*opt.Value); err == nil && connector > val {
+						err = fmt.Errorf("connector %d exceeds max available connectors: %d", connector, val)
 					}
+
+				case ocpp.KeySupportedFeatureProfiles:
+					if !strings.Contains(*opt.Value, "SmartCharging") {
+						err = fmt.Errorf("the mandatory SmartCharing profile is not supported")
+					}
+					c.hasRemoteTriggerFeature = strings.Contains(*opt.Value, "RemoteTrigger")
```

Comment:
Könnte man hier die ocpp Konstante nutzen?

Reply:
Hab zumindest spontan nichts passendes gefunden. In types.go gibt es dazu nichts.

Reply:
`smartcharging.ProfileName`

Reply:
Aaah, Si!